source/puppet/4.2/reference/ssl_attributes_extensions.markdown

```diff
@@ -171,53 +174,51 @@ Puppet Labs provides several registered OIDs (under "ppRegCertExt") for the most
 
 {% partial ./_registered_oids.md %}
 
-AWS Attributes and Extensions Population Example
------
+## AWS Attributes and Extensions Population Example
 
-Generally, you will want to use an automated script (possibly using cloud-init or a similar tool) to populate the `csr_attributes.yaml` file at the time a node is provisioned.
+You can use an automated script (possibly using cloud-init or a similar tool) to populate the `csr_attributes.yaml` file when you provision a node.
 
-As an example, you could enter the following script into the "Configure Instance Details —> Advanced Details" section when provisioning a new node from the AWS EC2 dashboard:
+As an example, you can enter the following script into the "Configure Instance Details —> Advanced Details" section when provisioning a new node from the AWS EC2 dashboard:
 
-    #!/bin/sh
-    if [ ! -d /etc/puppet ]; then
-       mkdir /etc/puppet
-    fi
-    erb > /etc/puppet/csr_attributes.yaml <<END
-    custom_attributes:
-      1.2.840.113549.1.9.7: mySuperAwesomePassword
-    extension_requests:
-      pp_instance_id: <%= %x{/opt/aws/bin/ec2-metadata -i}.sub(/instance-id: (.*)/,'\1').chomp %>
-      pp_image_name:  <%= %x{/opt/aws/bin/ec2-metadata -a}.sub(/ami-id: (.*)/,'\1').chomp %>
-    END
+~~~ bash
+#!/bin/sh
+if [ ! -d /etc/puppetlabs/puppet ]; then
+   mkdir /etc/puppetlabs/puppet
+fi
+erb > /etc/puppetlabs/puppet/csr_attributes.yaml <<END
+custom_attributes:
+  1.2.840.113549.1.9.7: mySuperAwesomePassword
+extension_requests:
+  pp_instance_id: <%= %x{/opt/aws/bin/ec2-metadata -i}.sub(/instance-id: (.*)/,'\1').chomp %>
+  pp_image_name:  <%= %x{/opt/aws/bin/ec2-metadata -a}.sub(/ami-id: (.*)/,'\1').chomp %>
+END
+~~~
 
-Assuming your image had the `erb` binary available, this would populate the attributes file with the AWS instance ID, image name, and a pre-shared key to use with policy-based autosigning.
+Assuming your image has the `erb` binary available, this populates the attributes file with the AWS instance ID, image name, and a pre-shared key to use with policy-based autosigning.
 
+## Troubleshooting
 
-Troubleshooting
------
+### Recovering from Failed Data Embedding
 
-### Recovering From Failed Data Embedding
+When first testing this feature, you might fail to embed the right information in a CSR or certificate and want to start over for your test nodes. (This is not really a problem once your provisioning system is changed to populate the data, but it can happen pretty easily when doing things manually.)
 
-When first testing this feature, you might fail to embed the right information in a CSR or certificate and want to start over for your test node(s). (This is not really a problem once your provisioning system is changed to populate the data, but it can happen pretty easily when doing things manually.)
+To start over, do the following:
 
-In order to start over, you'll need to do the following:
-
-**On the test node:**
+**On the test node**:
```

Comment:
Not going to block on it, but this is typographically wrong; punctuation adjacent to text should have the same font styling as the text it adjoins.